backend/tests/getOrCreateSession.test.ts

```diff
@@ -0,0 +1,134 @@
+import { describe, expect, it, jest } from '@jest/globals';
+import { Archive } from 'opentok';
+import mockOpentokConfig from '../helpers/__mocks__/config';
+
+await jest.unstable_mockModule('../helpers/config', mockOpentokConfig);
+
+const mockSessionsInStorage: { [key: string]: string } = {};
+const mockVcrSessionStorage = {
+  getSession: (key: string) => mockSessionsInStorage[key],
+  setSession: (key: string, sessionId: string) => {
+    mockSessionsInStorage[key] = sessionId;
+  },
+};
+
+await jest.unstable_mockModule('../storage/inMemorySessionStorage.ts', () => {
+  return {
+    default: jest.fn().mockImplementation(() => mockVcrSessionStorage),
+  };
+});
+
+const createSessionMock = jest.fn<() => Promise<{ sessionId: string }>>().mockResolvedValue({
+  sessionId: 'some-session',
+});
+
+await jest.unstable_mockModule('../videoService/opentokVideoService.ts', () => {
+  return {
+    default: jest.fn().mockImplementation(() => {
+      return {
+        startArchive: jest.fn<() => Promise<string>>().mockResolvedValue('archiveId'),
+        stopArchive: jest.fn<() => Promise<string>>().mockRejectedValue('invalid archive'),
+        generateToken: jest
+          .fn<() => Promise<{ token: string; apiKey: string }>>()
+          .mockResolvedValue({
+            token: 'someToken',
+            apiKey: 'someApiKey',
+          }),
+        createSession: createSessionMock,
+        listArchives: jest
+          .fn<() => Promise<Archive[]>>()
+          .mockResolvedValue([{ id: 'archive1' }, { id: 'archive2' }] as unknown as Archive[]),
+      };
+    }),
+  };
+});
+
+const { getOrCreateSession } = await import('../routes/session');
+
+describe('getOrCreateSession', () => {
```

Comment:
The reason this is not in session.test.ts is that those tests use supertest to test the express endpoints, but those requests do not run in parallel like in real life, even if you don't await them, they run one by one.